### PR TITLE
[Feature] Add skill hero block

### LINKED ISSUE #6938: ✨ Add a "Skill showcase" hero block to the "Profile and applications" page
## ✨ Details ([Figma file](https://www.figma.com/file/jHsXzqMfBtomLa1Gom94Y7/Candidates---Profile-and-applications?type=design&node-id=1097-34868))

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #6938

## 👋 Introduction

This branch adds a new skills block to the Profile and applications page.

## 🕵️ Details

Keep in mind:
1. This block is totally non-functional.  None of the links or counters work yet.
2. This block only appears when the FEATURE_SKILL_LIBRARY flag is set.

## 🧪 Testing

1. Build the app.
2. Navigate to `http://localhost:8000/en/applicant/dashboard`.
3. Compare the block to the design.
4. Double-check that the page looks OK when the flag is turned off.

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx [hunk 1]
```diff
@@ -98,6 +101,26 @@ const DashboardHeading = ({ user }: DashboardHeadingProps) => {
     notEmptyExperiences?.filter(isPersonalExperience) || [];
   const workExperiences = notEmptyExperiences?.filter(isWorkExperience) || [];
 
+  // TODO - complete these
```
Comment (petertgiles) [inline apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx:104]: Stubs for future logic and links.

**Review by petertgiles — COMMENTED** [review]
A few comments.

File: apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx [hunk 2]
```diff
@@ -240,8 +263,16 @@ const DashboardHeading = ({ user }: DashboardHeadingProps) => {
       )}
       <div
         data-h2-display="base(grid)"
-        data-h2-grid-template-columns="base(100%) p-tablet(repeat(2, minmax(0, 1fr)))"
         data-h2-gap="base(x1)"
+        {...(skillLibraryFlag
+          ? {
+              "data-h2-grid-template-columns":
+                "base(100%) p-tablet(repeat(3, minmax(0, 1fr)))",
```
Comment (esizer) [inline apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx:270]: We may want to map this `l-tablet` this is cutting it close 😅 What do you think @substrae ?

Screenshot 2023-06-23 145449

**Review by esizer — CHANGES_REQUESTED** [review]
Looking really great so far! 👍 I just noticed the breakpoint may need some adjusting.

File: apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx (hunk 2 above)
Comment (substrae) [reply apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx:270]: @esizer

Oof, yeah, I think `l-tablet` is a better choice. Good catch!

Comment (petertgiles) [reply apps/web/src/pages/Applications/ApplicantDashboardPage/components/DashboardHeading.tsx]: Yeah, good call.  This is minimum size now.
image
Ref: [change wrap size](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7089/commits/1ddad25380fcfdc052fa105f03f307b876aeed7b)

**Review by esizer — APPROVED** [review]
Awesome! ⭐

**Comment by substrae** [comment]
I'm assuming I can't review this in Storybook because it's behind a feature flag? Possible I'm just not looking hard enough though 😅

**Comment by petertgiles** [comment]
> I'm assuming I can't review this in Storybook because it's behind a feature flag?

Yeah, it's hidden behind a feature flag and right now we don't enable flags in storybook.  I'm going to make an issue to make them controllable.  In the meantime, if you like, I could deploy it to the dev server.

**Comment by substrae** [comment]
@petertgiles 
 
No need, I can pull the branch down and take a look that way. +1 for feature flag toggles in Storybook though 🤩

**Comment by substrae** [comment]
@petertgiles 

Couple small tweaks for ya:
- We can probably ditch the color transition on the card header links - the underline disappearing and the arrows moving is enough I think
- Items in the cards that aren't links are still changing color on hover, which is a tad confusing

Otherwise, looks good to me!

**Comment by petertgiles** [comment]
> Couple small tweaks for ya:

Thanks!  First point fixed in [removed common link styles](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7089/commits/48a0126a0930d279523c40e47daffdd077da43d8) and second point in [remove hover for status item](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7089/commits/aea1c88a995ea8b0f555512b14288bcb78e4ecd2).

**Pull request merged**